ray-operator/controllers/ray/common/pod.go

```diff
@@ -141,7 +166,9 @@ func BuildPod(podTemplateSpec v1.PodTemplateSpec, rayNodeType rayiov1alpha1.RayN
 		// replacing the old command
 		pod.Spec.Containers[rayContainerIndex].Command = []string{"/bin/bash", "-c", "--"}
 		if cmd != "" {
-			args = fmt.Sprintf("%s && %s", cont, cmd)
+			// If 'ray start' has --block specified, commands after it will not get executed.
+			// so we need to put cmd before cont.
```

Comment:
The order might matter for some users, so we might breaking backwards compatibility in some cases. It's probably not worth worrying about, though.

Notes for the future --

IMO, we should always block, no matter what. Executing a container entrypoint in a subprocess and sleeping is a weird pattern.

I find the way the user's cmd and args are processed in this method a little confusing. We should make sure to document the behavior if we haven't already.